pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -157,10 +157,9 @@ func (e *DockerRegistryServiceController) waitForDockerURLs(ready chan<- struct{
 		return
 	}
 
-	// after syncing, determine the current state and assume that we're up to date for it if you don't do this,
-	// you'll get an initial storm as you mess with all the dockercfg secrets every time you startup
+	// after syncing, make sure the dockercfgController is up to date before releasing it
+	// this controller will do a single scan of all existing secrets to be sure that they're up to date
 	urls := e.getDockerRegistryLocations()
-	e.setRegistryURLs(urls...)
```

Comment:
@mfojtik this is the line you found before.  Removing this causes us to rescan all secrets once and then only on changes thereafter, right?

Reply:
@deads2k gotcha, this will cause `e.getRegistryURLs` to return empty map on the first run which will make https://github.com/openshift/origin/pull/19791/files#diff-f9db327ed89bae5fe5de02a82f7a91d8R244 pass.